src/Microsoft.AspNetCore.SystemWebAdapters/Adapters/RemoteAppExtensions.Shared.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+namespace Microsoft.AspNetCore.SystemWebAdapters;
+
+public static class RemoteAppExtensions
+{
+    /// <summary>
+    /// Add configuration for connecting to a remote app for System.Web extensions that require a remote
+    /// ASP.NET app such as remote app authentication or remote app session sharing.
+    /// </summary>
+    public static ISystemWebAdapterBuilder AddRemoteApp(this ISystemWebAdapterBuilder builder, Action<RemoteAppOptions> configure)
+    {
+        if (builder is null)
+        {
+            throw new ArgumentNullException(nameof(builder));
+        }
+
+        if (configure is null)
+        {
+            throw new ArgumentNullException(nameof(configure));
+        }
+
+        var options = builder.Services.AddOptions<RemoteAppOptions>()
+            .Configure(configure);
+
+#if NET6_0_OR_GREATER
+        options.ValidateDataAnnotations();
+#endif
```

Comment:
```suggestion
        builder.Services.AddOptions<RemoteAppOptions>()
#if NET6_0_OR_GREATER
            .ValidateDataAnnotations();
#endif
            .Configure(configure);

```

Reply:
Updated (though I didn't take the suggestion directly since there's an extra ; in it).